dashboard/src/lib/auth.ts

```diff
@@ -89,9 +90,16 @@ export const authOptions: NextAuthOptions = {
         token.role = user.role;
         token.totpEnabled = user.totpEnabled;
         token.hasPassword = Boolean((user as unknown as { passwordHash?: string | null })?.passwordHash);
+        token.onboardingCompletedAt =
+          (user as unknown as { onboardingCompletedAt?: Date | null })?.onboardingCompletedAt ?? null;
+        token.termsAcceptedAt = (user as unknown as { termsAcceptedAt?: Date | null })?.termsAcceptedAt ?? null;
+        token.termsAcceptedVersion =
+          (user as unknown as { termsAcceptedVersion?: number | null })?.termsAcceptedVersion ?? null;
 
         if (account?.provider === 'google') {
-          const emailVerifiedByGoogle = Boolean((profile as any)?.email_verified);
+          const emailVerifiedByGoogle = Boolean(
+            (profile as unknown as { email_verified?: boolean })?.email_verified,
+          );
```

Comment:
This seems to only include `google` provider. We don't have a case for `github` for instance. Is that necessary?